pkgs/os-specific/darwin/install-binary-package/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenvNoCC, makeSetupHook, _7zz, libarchive }:
+
+{ appName ? "."
+, nativeBuildInputs ? []
+, sourceRoot ? "."
+, ...
+}@args:
+
+let
+  unpackDmgPkg = makeSetupHook {
+    name = "unpack-dmg-pkg";
+    propagatedBuildInputs = [ _7zz libarchive ];
+  } ./unpack-dmg-pkg.sh;
+in
+stdenvNoCC.mkDerivation (args // {
```

Comment:
```suggestion
let
  unpackDmgPkg = makeSetupHook {
    name = "unpack-dmg-pkg";
    propagatedBuildInputs = [ _7zz libarchive ];
  } ./unpack-dmg-pkg.sh;
in
(stdenvNoCC.mkDerivation fpArgs).overrideAttrs (finalAttrs:

{ appName ? "."
, nativeBuildInputs ? []
, sourceRoot ? "."
, ...
}@args:

{
```

Reply:
Thanks for this, @ShamrockLee, I've included your suggestion locally, but need to refactor the treewide changes before pushing an update on this.
Also I'd like wait until `sourceRoot` conversation (see [below](https://github.com/NixOS/nixpkgs/pull/293498#discussion_r1531438958)) provides guidance on a more acceptable solution.